### ISSUE TITLE
Panic running board

### ISSUE THREAD

**Issue by jdm** [jdm/libmzx#79]
https://www.digitalmzx.com/show.php?id=1494

```
thread 'main' panicked at 'index out of bounds: the len is 3025 but the index is 3269', /Users/jdm/src/libmzx/src/lib.rs:336:10
stack backtrace:
   0: rust_begin_unwind
             at /rustc/1c389ffeff814726dec325f0f2b0c99107df2673/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/1c389ffeff814726dec325f0f2b0c99107df2673/library/core/src/panicking.rs:92:14
   2: core::panicking::panic_bounds_check
             at /rustc/1c389ffeff814726dec325f0f2b0c99107df2673/library/core/src/panicking.rs:69:5
   3: <usize as core::slice::index::SliceIndex<[T]>>::index
             at /Users/jdm/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/index.rs:182:10
   4: core::slice::index::<impl core::ops::index::Index<I> for [T]>::index
             at /Users/jdm/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/index.rs:15:9
   5: <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index
             at /Users/jdm/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/vec.rs:2161:9
   6: libmzx::Board::level_at
             at /Users/jdm/src/libmzx/src/lib.rs:336:10
   7: libmzx::robot::run_one_command
             at /Users/jdm/src/libmzx/src/robot.rs:1039:60
   8: libmzx::robot::update_robot
             at /Users/jdm/src/libmzx/src/robot.rs:469:15
   9: libmzx::board::update_board
             at /Users/jdm/src/libmzx/src/board.rs:189:34
  10: capture::run
             at ./capture/src/main.rs:145:17
  11: capture::main
             at ./capture/src/main.rs:223:9
  12: core::ops::function::FnOnce::call_once
             at /Users/jdm/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
```

**Comment by jdm** [comment]
https://github.com/jdm/libmzx/blob/7d38fc0147448770b6eb3a15c995b92ec8e6ae99/src/robot.rs#L1040

**Comment by jdm** [comment]
I guess level_at and all similar APIs need to either Options, because the coordinates can always be outside of the actual board size.

**Comment by jdm** [comment]
Fixed by 615dae20f2d82a2bab6d19cb089ada5225b55d56
